From e38da4eb088c2562e1d073a0ad83346e4ad2525a Mon Sep 17 00:00:00 2001
From: Jeroen Hofstee <jhofstee@victronenergy.com>
Date: Fri, 9 Dec 2016 14:36:06 +0100
Subject: [PATCH 1/2] Use MemAvailable instead of MemFree

MemFree does not include cached pages and hence underestimates the
amount of memory that _could_ be freed when needed. MemFree + Cached
is a better estimate, but it's not quite right either since Cached
includes things that can't be freed.  For this reason, kernels since 3.14
include a MemAvailable field which provides a more accurate approximation.
See https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/patch/?id=34e431b0ae398fc54ea69ff85ec700722c9da773

Note: This patch blindly assumes MemAvailable is present!!!
Completely unsuitable for general use, since the device ends up in a
reboot loop in such a case...
---
 src/memory.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/memory.c b/src/memory.c
index ff2489ac1d8c..6b9b99c225a0 100644
--- a/src/memory.c
+++ b/src/memory.c
@@ -28,7 +28,7 @@
 #include "extern.h"
 #include "watch_err.h"
 
-#define FREEMEM		"MemFree:"
+#define FREEMEM		"MemAvailable:"
 #define FREESWAP	"SwapFree:"
 
 static int mem_fd = -1;
-- 
2.26.2

